monkelib/API/get_data.py

```diff
@@ -99,47 +102,52 @@ def get_user_repo(username: str) -> bool:
         return False
 
 
-def get_most_starred_python_repos() -> bool:
+def get_most_starred_repos() -> bool:
     """
-    This function gets the most starred Python repositories from the GitHub API.
+    This function gets the most starred repositories for multiple languages from the GitHub API.
     :return: True if the request was successful, False if not.
     """
 
     url: str = 'https://api.github.com/search/repositories'
-    query_params: dict = {
-        'q': 'language:python',
-        'per_page': 10  # Number of results per page (max 100)
-    }
-    # Load the token from the pat.json file
+
+    queryList = []
+    for language in LANGUAGE_LIST:
+        queryList.append({
+            'q': f'language:{language}',
+            'per_page': 75,
+        })
+
     token: str = _load_secret()
 
     headers: dict = {
         'Authorization': f'token {token}',
         'Accept': 'application/vnd.github.v3+json'
     }
 
-    try:
-        response: requests.Response = requests.get(url, headers=headers, params=query_params)
-        response.raise_for_status()  # This will raise an exception for HTTP error codes
-
-        repos_data: list = []
-
-        for repo in response.json()['items']:
-            repo_info = _get_repo_fields(repo)
-            repos_data.append(repo_info)
-
-        # Writing data to a JSON file
-        _save_to_json(repos_data, 'most_starred_python_repos.json')
-        print("Data successfully written to most_starred_python_repos.json")
-
-        # Writing data to a CSV file
-        _save_to_csv(repos_data, 'most_starred_python_repos.csv')
-        print("Data successfully written to most_starred_python_repos.csv")
-        return True
-
-    except requests.exceptions.HTTPError as err:
-        print(f"HTTP error occurred: {err}")
-        return False
-    except Exception as err:
-        print(f"An error occurred: {err}")
-        return False
+    dataList = []
+    for query in queryList:
+        try:
+            response: requests.Response = requests.get(url, headers=headers, params=query, allow_redirects=False)
+            response.raise_for_status()
+
+            repos_data: list = []
+
+            for repo in response.json()['items']:
+                repo_info = _get_repo_fields(repo)
+                repos_data.append(repo_info)
+
+            dataList.append(repos_data)
+        except requests.exceptions.HTTPError as err:
+            print(f"HTTP error occurred: {err}")
+            return False
+        except Exception as err:
+            print(f"An error occurred: {err}")
+            return False
+
+    dataList = [item for sublist in dataList for item in sublist]   # Flatten the list
+    _save_to_json(dataList, 'mostStarredRepos.json')
+    print("Data successfully written to mostStarredRepos.json")
+
+    _save_to_csv(dataList, 'mostStarredRepos.csv')
+    print("Data successfully written to mostStarredRepos.csv")
```

Comment:
Why are we exporting to both json and to csv ? Pretty sure we just need csv

Reply:
I think its good to store both. We can have the raw data and our transformed data
![Uploading Guy-Looking-At-Server.jpg…]()

Reply:
While I'm sure we are always going to work with just the CSV I was asked to save both at the start, and then we could remove this